scripts/build-layout-scales.js

```diff
@@ -0,0 +1,1271 @@
+      `);
+    }, '');
+    css += '/** @endgroup */\n';
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--m-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .z${scale}-mm {
+          z-index: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--l-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .z${scale}-ml {
+          z-index: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--xl-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .z${scale}-mxl {
+          z-index: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    return css;
+  };
+
+  variantGenerators.multiMargin = function (scales) {
+    function value(v) {
+      return `${v}px`;
+    }
+
+    let css = stripIndent(`
+    \n/**
+       * Apply margin on all sides.
+       *
+       * @group
+       * @example
+       * <div class='m24 bg-darken10'>m24</div>
+       * @memberof Margins
+       */`
+    );
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .m${scale} {
+          margin: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += '/** @endgroup */\n';
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--m-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .m${scale}-mm {
+          margin: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--l-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .m${scale}-ml {
+          margin: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--xl-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .m${scale}-mxl {
+          margin: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`
+    \n/**
+      * Apply margin on the top and bottom.
+      *
+      * @group
+      * @example
+      * <div class='my24 bg-darken10'>my24</div>
+      * @memberof Margins
+      */`
+    );
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .my${scale} {
+          margin-top: ${value(scale)} !important;
+          margin-bottom: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += '/** @endgroup */\n';
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--m-screen']} {`);
```

Comment:
This might be a totally menial optimization, but we *could*, if we were interested, group media queries pretty easily with this build script. Instead of adding rules into these individual queries you could push them into a an array (`mediumScreenRules`) and then build one media query block after you've run through all these.